src/main/kotlin/com/expedia/graphql/schema/SchemaGenerator.kt

```diff
@@ -297,4 +304,22 @@ internal class SchemaGenerator(
 
         return interfaceType
     }
+
+    private fun unionType(kClass: KClass<*>): GraphQLType {
+        val builder = GraphQLUnionType.newUnionType()
+
+        builder.name(kClass.simpleName)
+        builder.description(kClass.graphQLDescription())
+
+        builder.typeResolver { env: TypeResolutionEnvironment -> env.schema.getObjectType(env.getObject<Any>().javaClass.simpleName) }
+
+        val reflections = Reflections(config.supportedPackages)
+        val implementations = reflections.getSubTypesOf(Class.forName(kClass.javaObjectType.name))
+        implementations.forEach {
+            val possibleType = objectType(it.kotlin, null)
```

Comment:
interfaces should be filtered out otherwise you'll end up in a bad state when a type implements another interface